### PR TITLE
chore(*): improve error messages for missing links

### PR DESCRIPTION
**Pull request by vados-cosmonic**
## Feature or Problem

This commit improves error messaging when we fail to call a target for an interface that has no link specified.

Resolves https://github.com/wasmCloud/wasmCloud/issues/1927

### CODE REVIEW

File: crates/host/src/wasmbus/handler.rs [hunk 1]
```diff
@@ -107,65 +107,113 @@ impl Handler {
...
-                "http",
-                "outgoing-handler",
+                ns, pkg, iface
             )))
             .await
-            .context("unknown `wasi:http/outgoing-handler` target")?;
+            .with_context(|| {
+                let mut msg = format!("failed to call interface `{ns}:{pkg}/{iface}`");
+                if let Ok(false) = self.try_link_exists_for_interface(ns, pkg, iface) {
+                    msg.push_str(&format!(" (failed to find a link from this component [{}], please check your configuration)", self.component_id));
```
Comment (brooksmtownsend) [inline crates/host/src/wasmbus/handler.rs:137]:
```suggestion
                    msg.push_str(&format!(" (failed to find a configured link from component [{}], please check your configuration)", self.component_id));
```

consistency with other messages suggestion

File: crates/host/src/wasmbus/handler.rs [hunk 2]
```diff
@@ -545,7 +593,13 @@ impl Bus for Handler {
             tx.await.context("failed to transmit parameters")?;
             Ok(results)
         } else {
-            bail!("component attempted to invoke a function on an unknown target")
+            bail!(
+                "component [{}] attempted to invoke a function [{}/{}] on an unknown target [{}]",
+                self.component_id,
+                instance,
+                name,
+                target.lattice_id().unwrap_or("<unknown>"),
```
Comment (brooksmtownsend) [inline crates/host/src/wasmbus/handler.rs:601]: Shouldn't this target be the target "id", rather than the lattice ID?

Comment (brooksmtownsend) [reply crates/host/src/wasmbus/handler.rs:601]: Or, after reading the implementation of the method, this is not the _lattice name_ this is the invocation target on the lattice, right? If I understand correctly this LGTM

**Review by brooksmtownsend — APPROVED** [review]
One request for consistency one ask for my own clarification

File: crates/host/src/wasmbus/handler.rs (hunk 1 above)
Comment (vados-cosmonic) [reply crates/host/src/wasmbus/handler.rs:137]: Ah thank you!

File: crates/host/src/wasmbus/handler.rs (hunk 2 above)
Comment (vados-cosmonic) [reply crates/host/src/wasmbus/handler.rs:601]: Ah this is not the lattice name -- it's the "reference by which the entity can be addressed on the lattice" becuase that is off the `TargetEntity`. 

This is confusing though, so I'll change this!

**Pull request merged**